src/operators/function.ts

```diff
@@ -9,9 +9,14 @@ export interface FunctionOperatorOptions extends OperatorOptions {
 /**
  * FunctionOperator executes a function and returns the result.
  */
-export class FunctionOperator extends Operator<FunctionOperatorOptions> {
+export class FunctionOperator extends Operator {
```

Comment:
Removing the generic type variable on the `Operator` class impacted this class the most. You can see how I have to include a "useless constructor" as well as explicitly cast the `options` instance when it is referenced.